internal/certs/certs_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certs
+
+import (
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestGenerateCASecret(t *testing.T) {
+	host1 := "134.565.56.77"
+	host2 := "172.345.82.7"
+	name := "ca-secret"
+	host := host1 + ", " + host2
+	cn := "www.example.com"
+	ca_secret := GenerateSecret(name, cn, host, 0, nil)
+	data, ok := ca_secret.Data["tls.crt"]
+	if !ok {
+		t.Error("Invalid secret, tls.crt is missing")
+	}
+	cert, err := DecodeCertificate(data)
+
+	if err != nil {
+		t.Error("Error decoding certificate")
+	}
+
+	notBefore := time.Now()
+	expiration := 5 * 365 * 24 * time.Hour
+	notAfter := notBefore.Add(expiration)
+	date_match := cert.NotAfter.Year() == notAfter.Year() && cert.NotAfter.YearDay() == notAfter.YearDay()
```

Comment:
Out of an abundance of caution, I'd suggest logic like this where we pick a window of time that we expect the certificate's expiry to be. Comparing calendar date and year is an opportunity for this to flake.

```
notAfterTolerance := time.Hour
assert.Assert(t, notAfter.Add(-1 * notAfterTolernance / 2).Before(cert.NotAfter))
assert.Assert(t, cert.NotAfter.Before(notAfter.Add(notAfterTolernance / 2)))
```